packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx

```diff
@@ -1,211 +1,81 @@
-    }
-
-    if (top > HEIGHT_THRESHOLD) {
-      newPosition.top = zeroSpacing;
-      newPosition.bottom = undefined;
-    }
-    return newPosition;
-  }
-
-  if (right > WIDTH_THRESHOLD) {
-    newPosition.right = zeroSpacing;
-    newPosition.left = undefined;
-  }
-
-  if (bottom > HEIGHT_THRESHOLD) {
-    newPosition.bottom = `${Number(triggererEl?.clientHeight) + Number(size[20])}px`;
-    newPosition.top = undefined;
-  }
-
-  if (top > HEIGHT_THRESHOLD) {
-    newPosition.top = zeroSpacing;
-    newPosition.bottom = undefined;
-  }
-
-  return newPosition;
-};
-
-const getDropdownOverflowMiddleware = ({
-  isMenu,
-  triggererRef,
-  actionListItemRef,
-  setDropdownPosition,
-}: {
-  setDropdownPosition: React.Dispatch<React.SetStateAction<DropdownPosition>>;
-  isMenu: boolean;
-  triggererRef: React.RefObject<HTMLButtonElement | null>;
-  actionListItemRef: React.RefObject<HTMLDivElement | null>;
-}): { name: string; fn: (state: MiddlewareState) => Promise<object> } => {
-  return {
-    name: 'detectOverflowMiddleware',
-    async fn(state: MiddlewareState) {
-      const overflow = await detectOverflow(state, {
-        elementContext: 'reference',
-      });
-      const position = getDropdownOverlayPosition({
-        overflow,
-        isMenu,
-        triggererEl: triggererRef.current,
-        actionListItemEl: actionListItemRef.current,
-      });
-      setDropdownPosition(position);
-      return {};
-    },
-  };
-};
+const OVERLAY_OFFSET: number = size['8'];
+const OVERLAY_PADDING: number = size['12']; // doesn't have to be exact. Just rough padding for floating ui to decide to show overlay on top or bottom
+const OVERLAY_ZINDEX = 1001;
 
 /**
  * Overlay of dropdown
  *
  * Wrap your ActionList within this component
  */
-const _DropdownOverlay = ({ children, testID }: DropdownOverlayProps): React.ReactElement => {
-  const {
-    isOpen,
-    triggererRef,
-    hasLabelOnLeft,
-    dropdownTriggerer,
-    setIsOpen,
-    actionListItemRef,
-  } = useDropdown();
+const _DropdownOverlay = ({
+  children,
+  testID,
+}: DropdownOverlayProps): React.ReactElement | null => {
+  const { isOpen, triggererRef, triggererWrapperRef, dropdownTriggerer, setIsOpen } = useDropdown();
   const { theme } = useTheme();
   const bottomSheetAndDropdownGlue = useBottomSheetAndDropdownGlue();
-  const [showFadeOutAnimation, setShowFadeOutAnimation] = React.useState(false);
-  const [width, setWidth] = React.useState<SpacingValueType>('100%');
-  const [dropdownPosition, setDropdownPosition] = React.useState<DropdownPosition>({});
 
   const isMenu = dropdownTriggerer !== 'SelectInput';
 
-  const { refs } = useFloating({
+  const { refs, floatingStyles, context } = useFloating({
     open: isOpen,
     onOpenChange: setIsOpen,
-    strategy: 'absolute',
+    strategy: 'fixed',
     placement: 'bottom-start',
     elements: {
-      reference: triggererRef.current,
+      reference: triggererWrapperRef.current ?? triggererRef.current,
```

Comment:
can we add a comment why triggererWrapperRef ?? triggerRef is needed?

Reply:
I think yeah add this in code
https://github.com/razorpay/blade/pull/1559#discussion_r1305438920

Reply:
done